frontend/src/concepts/k8s/NameDescriptionField.tsx

```diff
@@ -1,8 +1,22 @@
 import * as React from 'react';
-import { FormGroup, Stack, StackItem, TextArea, TextInput, Tooltip } from '@patternfly/react-core';
+import {
+  FormGroup,
+  Stack,
+  StackItem,
+  TextArea,
+  TextInput,
+  Tooltip,
+  HelperText,
+  HelperTextItem,
+  HelperTextItemProps,
+} from '@patternfly/react-core';
 import { HelpIcon } from '@patternfly/react-icons';
 import { NameDescType } from '~/pages/projects/types';
-import { isValidK8sName, translateDisplayNameForK8s } from '~/pages/projects/utils';
+import {
+  isValidK8sName,
+  translateDisplayNameForK8s,
+  isValidK8sNameLength,
+} from '~/pages/projects/utils';
```

Comment:
We want to avoid dependencies in the direction of `components` -> `pages`.
See [coding structure](https://github.com/opendatahub-io/odh-dashboard/blob/main/docs/architecture.md#coding-structure) architecture docs.

K8s resources may conform to different naming rules and therefore such validation rules should be passed in as props. It may be ok to define a singular rule set as the default. I do not believe the intention here is also to change the validation of project names. However that's exactly what's happening.